### PR TITLE
Add a migration doc for Android 1.x to 2.0

### PR DESCRIPTION
**Pull request by yangyansong-adbe**
This PR creates a `Migrate to latest Android SDK` section/page for existing customers who want to migrate to Android SDK 2.0. This is an initial PR for Andriod migration documentation. It contains the migration overview and detailed migration steps for some extensions (updates for other extensions will be in future PRs).

### CODE REVIEW

File: src/pages/documentation/migrate-to-android.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+## Replace the deprecated API calls with the new API calls
+
+Several Android 1.x APIs are depreacted in Android 2.0 libraries, you can choose to replace the deprecated API calls in your code with calling the alternative APIs:
+
+| Deprecated API                  | Alternative API                  |
+| :------------------------------ | :------------------------------- |
+| Identity.registerExtension      | MobileCore.registerExtensions    |
+| Signal.registerExtension        |                                  |
+| Lifecycle.registerExtension     |                                  |
+| UserProfile.registerExtension   |                                  |
```
Comment (calebk1) [inline src/pages/documentation/migrate-to-android.md:87]: Do these not have alternative APIs? I'd assume it's just the MobileCore.registerExtensions, but I want to be doubly sure.

Also, would be good IMO to wrap the APIs in the `` ticks for code lines.

(Also same comment as above for table spacing/whitespace.)

**Review by calebk1 — COMMENTED** [review]
A few comments! Looking good though.

File: src/pages/documentation/migrate-to-android.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+<InlineAlert variant="warning" slots="text"/>
+
+Since we will not provide a `Core bundle` artifact automatically including Core, Signal, Lifecycle and Identity artifacts, you need to manually import those libraries in Android SDK 2.0.
+
+| Android SDK 1.x | Android SDK 2.0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
```
Comment (emdobrin) [inline src/pages/documentation/migrate-to-android.md:19]: The naming may be conflicting with the versions of Android. Let's rename first column to `Extension name`, second to `Latest version`

File: src/pages/documentation/migrate-to-android.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+# Migrate to latest Android SDK
+
+If you have implemented the Android SDK 1.x versions, then this guide will help you understand the steps required to migrate your implementation to the latest Android SDK versions (2.0 or higher). In summary, you'll need to:
+
+1. [Switch Gradle dependencies from 1.x to 2.0](#switch-gradle-dependencies)
+2. [Update SDK initialization](#update-sdk-initialization)
+3. [Replace the deprecated API calls with the new API calls](#replace-the-deprecated-api-calls-with-the-new-api-calls1)
+
+## Switch Gradle dependencies
```
Comment (emdobrin) [inline src/pages/documentation/migrate-to-android.md:11]: `Switch Gradle dependencies to latest versions`
Also use the same naming in the linked steps above

File: src/pages/documentation/migrate-to-android.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+### Manual library import
+
+If you are manually importing SDK libraries, ensure you identify all currently used Android 1.x libraries and switch them over to Android 2.0 libraries. The list of current Android 2.0 libraries can be found in the [current SDK versions document](sdk-versions.md#android) (in the Android section).
+
+### Gradle
+
+If you are using Gradle to manage your Adobe Experience Platform Mobile SDK dependencies, the following example shows you how to switch Android 1.x libraries to Android 2.0 libraries in your `build.gradle` file.
```
Comment (emdobrin) [inline src/pages/documentation/migrate-to-android.md:38]: If you are using Gradle to manage your app dependencies, the following example shows how to upgrade to latest version in the `build.gradle` file.

File: src/pages/documentation/migrate-to-android.md (hunk 1 above)
Comment (yangyansong-adbe) [reply src/pages/documentation/migrate-to-android.md:87]: Yes, those `registerExtension` APIs should be replaced by the`registerExtensions` API. Ideally, we can merge those empty cells, but I'm not sure if it's supported. Let me know if you have any suggestions.

File: src/pages/documentation/migrate-to-android.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    implementation 'com.adobe.marketing.mobile:signal:2.+'
+
+    // replace UserProfile 1.+ with UserProfile 2.0
+    //implementation 'com.adobe.marketing.mobile:userprofile:1.+'
+    implementation 'com.adobe.marketing.mobile:userprofile:2.+'
+
+}
+```
+
+<InlineAlert variant="warning" slots="text"/>
```
Comment (emdobrin) [inline src/pages/documentation/migrate-to-android.md:56]: can we move this one before the code sample for visibility?

File: src/pages/documentation/migrate-to-android.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+| CampaignClassic | [![Maven Central](https://img.shields.io/maven-central/v/com.adobe.marketing.mobile/campaignclassic.svg?logo=android&logoColor=white&label=campaignclassic)](https://mvnrepository.com/artifact/com.adobe.marketing.mobile/campaignclassic)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |
+
+### Manual library import
+
+If you are manually importing SDK libraries, ensure you identify all currently used Android 1.x libraries and switch them over to Android 2.0 libraries. The list of current Android 2.0 libraries can be found in the [current SDK versions document](sdk-versions.md#android) (in the Android section).
```
[the remark this answers is not in the document]
Comment (yangyansong-adbe) [reply src/pages/documentation/migrate-to-android.md:34]: @emdobrin  any suggestions/comments on this one?

File: src/pages/documentation/migrate-to-android.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+<InlineAlert variant="warning" slots="text"/>
+
+Warning Using dynamic dependency versions is not recommended for production apps. Refer to this [page](https://github.com/adobe/aepsdk-core-android/blob/main/Documentation/MobileCore/gradle-dependencies.md) for managing gradle dependencies.
+
+Save the `build.gradle` file and click `Sync Project with Gradle Files` in Android Studio to download the latet Android SDK.
+
+## Update SDK initialization
+
+After you have imported the new Android libraries, you'll need to update SDK initialization code as described below. With Android 2.0, the `MobileCore.start()` API is no longer required. The SDK has simplified initialization and registration of extensions by providing the `MobileCore.registerExtensions()` API. After the given extensions have been registered, the SDK will be initialized and the completion block will be executed. Code which used to reside in the start() block will now reside in the `MobileCore.registerExtensions()`'s completion block.
```
Comment (emdobrin) [inline src/pages/documentation/migrate-to-android.md:64]: With Mobile Core version 2.0.0 and above, the `MobileCore.start()`...

File: src/pages/documentation/migrate-to-android.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+The following code snippets show the new and correct initialization code required for the Android 2.0 libraries.
```
Comment (emdobrin) [inline src/pages/documentation/migrate-to-android.md:66]: ... show the recommended initialization code for the latest Mobile SDKs.

File: src/pages/documentation/migrate-to-android.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (emdobrin) [reply src/pages/documentation/migrate-to-android.md:34]: We probably don't need to link out of this document, I recommended rephrasing to something like this:

If you are manually importing the Mobile SDK libraries, update your app dependencies to the latest versions shown in the table above.

File: src/pages/documentation/migrate-to-android.md (hunk 1 above)
Comment (calebk1) [reply src/pages/documentation/migrate-to-android.md]: Ahh, I don't believe you can, unfortunately, for markdown tables. As a result, the best thing probably _is_ to be redundant and put it in each of the same tabs.

**Comment by yangyansong-adbe** [comment]
@calebk1 about those whitespaces and extra dashes (as below), they are automatically generated by a `code formatter` tool in VSCode. I don't see any drawbacks to this formatted table content. Can we just keep it?  
```
| Deprecated API                    | Alternative API                    |
| :-------------------------------- | :--------------------------------- |
| `Identity.registerExtension`      | `MobileCore.registerExtensions`    |
| `Signal.registerExtension`        |                                    |
| `Lifecycle.registerExtension`     |                                    |
| `UserProfile.registerExtension`   |                                    |
| `UserProfile.updateUserAttribute` | `UserProfile.updateUserAttributes` |
| `UserProfile.removeUserAttribute` | `UserProfile.removeUserAttributes` |
```

File: src/pages/documentation/migrate-to-android.md (hunk 1 above)
Comment (yangyansong-adbe) [reply src/pages/documentation/migrate-to-android.md]: @emdobrin  this PR doesn't contain migration details for all extensions. Can we add those in other PRs?

File: src/pages/documentation/migrate-to-android.md (hunk 5 above)
Comment (yangyansong-adbe) [reply src/pages/documentation/migrate-to-android.md:56]: I intend on showing the alert here since the following table shows `sdk-core` should be replaced with `core/identity/lifecycle/signal`.

File: src/pages/documentation/migrate-to-android.md (hunk 3 above)
Comment (emdobrin) [reply src/pages/documentation/migrate-to-android.md]: Can you also update the text for step 1 above to match this new title?

File: src/pages/documentation/migrate-to-android.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+<Tabs query="platform=java"/>
+
+Kotlin
+
+<Tabs query="platform=kotlin"/>
+
+## Replace the deprecated API calls with the new API calls
+
+Several Android 1.x APIs are depreacted in Android 2.0 libraries, you can choose to replace the deprecated API calls in your code with calling the alternative APIs:
```
Comment (emdobrin) [inline src/pages/documentation/migrate-to-android.md:80]: Android 1.x/2.x may conflict with android versions, so let's rename all occurrences:
Android 1.x -> previous versions of the Mobile SDK for Android
Android 2.0/2.x -> latest version of the Mobile SDK for Android

This example would rephrase to:
Some of the APIs available in previous major versions of the Mobile SDK for Android are now deprecated. You can choose to replace the deprecated APIs in your code with the alternative APIs in the latest version:

File: src/pages/documentation/migrate-to-android.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    // replace UserProfile 1.+ with UserProfile 2.0
+    //implementation 'com.adobe.marketing.mobile:userprofile:1.+'
+    implementation 'com.adobe.marketing.mobile:userprofile:2.+'
+
+}
+```
+
+<InlineAlert variant="warning" slots="text"/>
+
+Using dynamic dependency versions is not recommended for production apps. Refer to this [page](https://github.com/adobe/aepsdk-core-android/blob/main/Documentation/MobileCore/gradle-dependencies.md) for managing Gradle dependencies.
```
Comment (emdobrin) [inline src/pages/documentation/migrate-to-android.md:58]: we now have this file in this repo, let's use relative link here

File: src/pages/documentation/migrate-to-android.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+2. [Update SDK initialization](#update-sdk-initialization)
+3. [Replace the deprecated API calls with the new API calls](#replace-the-deprecated-api-calls-with-the-new-api-calls1)
+
+## Switch Gradle dependencies to latest versions
+
+At this time, the following Android SDK 1.x libraries may be switched out with their respective Android SDK 2.0 libraries. See instructions on proceeding further if you have [manually imported SDK libraries](#manual-library-import) , or if you used [Gradle to manage SDK dependencies](#Gradle).
+
+<InlineAlert variant="warning" slots="text"/>
+
+Since we will not provide a `Core bundle` artifact automatically including Core, Signal, Lifecycle and Identity artifacts, you need to manually import those libraries in Android SDK 2.0.
```
Comment (emdobrin) [inline src/pages/documentation/migrate-to-android.md:17]: let's make this sentence impersonal
Starting with version 2.0.0, the `sdk-core` bundle (which includes Core, Lifecycle, Identity, Signal) will no longer receive updates. You need to import those libraries individually as described below.

File: src/pages/documentation/migrate-to-android.md (hunk 10 above)
Comment (yangyansong-adbe) [reply src/pages/documentation/migrate-to-android.md:58]: Can you point me to the right page?

File: src/pages/documentation/migrate-to-android.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+<InlineAlert variant="warning" slots="text"/>
+
+Using dynamic dependency versions is not recommended for production apps. Refer to this [page](https://github.com/adobe/aepsdk-core-android/blob/main/Documentation/MobileCore/gradle-dependencies.md) for managing Gradle dependencies.
+
+Save the `build.gradle` file and select `Sync Project with Gradle Files` in Android Studio to download the latet Android SDK.
+
+## Update SDK initialization
+
+After you have imported the new Android libraries, you'll need to update SDK initialization code as described below. With Mobile Core version 2.0.0 and above, the `MobileCore.start()` API is no longer required. The SDK has simplified initialization and registration of extensions by providing the `MobileCore.registerExtensions()` API. After the given extensions have been registered, the SDK will be initialized and the completion block will be executed. Code which used to reside in the start() block will now reside in the `MobileCore.registerExtensions()`'s completion block.
```
Comment (emdobrin) [inline src/pages/documentation/migrate-to-android.md:64]: ... you'll need to update **the** SDK initialization code ...
**The code** which used to reside ...

File: src/pages/documentation/migrate-to-android.md (hunk 10 above)
Comment (emdobrin) [reply src/pages/documentation/migrate-to-android.md:58]: Not in this repo, it is in the latest version.. I guess we can use a direct link to https://developer.adobe.com/client-sdks/documentation/manage-gradle-dependencies

**Review by swarna04 — APPROVED** [review]
A couple more feedback comments to address before we merge. Thank you for creating this migration doc @yangyansong-adbe

File: src/pages/documentation/tabs/migrate-to-android.md [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+
+```java
+public class MainApp extends Application {
+    private static final String ENVIRONMENT_FILE_ID = "ENVIRONMENT_FILE_ID";
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        MobileCore.setApplication(this);
+        MobileCore.setLogLevel(LoggingMode.ERROR);
```
[the remark this answers is not in the document]
Comment (emdobrin) [reply src/pages/documentation/tabs/migrate-to-android.md:11]: @swarna04 can we keep this one to error here since it is the recommended setting for a production environment

[the remark this answers is not in the document]
Comment (swarna04) [reply src/pages/documentation/tabs/migrate-to-android.md:11]: hmm I think customers can use their own judgement if they want to enable warning or error for logging mode in their production apps. I'm fine if we want to demonstrate that here, however let's at least be consistent in the Java/ Kotlin examples. Kotlin example set's the log level to `VERBOSE` while it's set to `ERROR` in Java.

File: src/pages/documentation/migrate-to-android.md (hunk 10 above)
Comment (yangyansong-adbe) [reply src/pages/documentation/migrate-to-android.md:58]: The link shows 404.

Comment (calebk1) [reply src/pages/documentation/migrate-to-android.md:58]: That link 404s for now since it's not live yet. Since that will be coming in at the same time the ACP repo is made live, we should be fine on that front.

**Review by emdobrin — APPROVED** [review]
Looks awesome @yangyansong-adbe :shipit:  don't forget to update the gradle dependencies list before merging
